pandas/core/series.py

```diff
@@ -253,7 +254,7 @@ def __init__(
                 else:
                     data = data.reindex(index, copy=copy)
                 data = data._data
-            elif isinstance(data, dict):
+            elif isinstance(data, abc.Mapping):
```

Comment:
this is an orthogonal change pls remove

Reply:
It is not orthogonal.
In the function `_map_values` in `core/base.py`, in case the mapper is a subclass of `collections.abc.Mapping` and isn't a `dict` with the `__missing__` property, we create a Series out of the mapper. In case it is `collections.abc.Mapping` and isn't a `dict`, the creation of the Series will fail if not for this change.

Reply:
Is this hit in a test case?

Reply:
It is in the test cases I added. If I remove this change, they will fail, as a Series is created from the mapper.